Sources/AWSLambdaRuntimeCore/Lambda+LocalServer.swift

```diff
@@ -376,15 +378,17 @@ private struct LambdaHttpServer {
                 return nil
             }
 
-            if let element = await self.popFirst() {
+            if let element = self.popFirst() {
+                // if there is an element in the buffer, dequeue it
                 return element
             } else {
                 // we can't return nil if there is nothing to dequeue otherwise the async for loop will stop
-                // wait for an element to be enqueued
-                return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<T, any Error>) in
-                    // store the continuation for later, when an element is enqueued
-                    self._continuation.withLock {
-                        $0 = continuation
+                // so, wait for an element to be enqueued
+                return try await withCheckedThrowingContinuation {
+                    (continuation: CheckedContinuation<T, any Error>) in
+                    self.mutex.withLock { mutexContent in
+                        // store the continuation for later, when an element is enqueued
+                        mutexContent.1 = continuation
```

Comment:
This is still racy: between "popFirst" returning nil and storing the continuation, somebody could have already put sth in the buffer. Really, to implement this correctly, I think you need to acquire the lock, check your invariants, apply the necessary state change, release the lock and then run side-effects.

Reply:
The invariant here is: There can only be a stored continuation if the buffer is empty. If the buffer is non-empty, there cannot be a stored continuation.